docs/compiling.rst

```diff
@@ -70,6 +70,17 @@ that is supported via a ``build_ext`` command override; it will only affect
         ext_modules=ext_modules
     )
 
+If you have single-file extension modules that are directly stored in the
```

Comment:
One feedback datapoint: after skimming through this paragraph I'm asking myself "uhm, what is this doing?"

Guessing: is foo.cpp compiled and linked as foo.so when running pip install?
If that's it, saying that first could make the rest easier to digest.

Similar to @EricCousineau-TRI, I don't have a lot of experience with setuptools.
But if this feature is well explained and tested I'd support adding it into pybind11.
Maybe with some small addition: For questions regarding this feature please tag @anntzer.

Reply:
I clarified the point as suggested.
I'm happy to be tagged if there's any question, but would rather not add a big pointer to myself in the docs :)

Reply:
Thanks! That's clear now.

What do you think about adding a `# source code comment` in setup_helpers.py, above or below the docstring, to tell people about the tagging?

There is a bigger general question about pybind11 maintenance here, which is beyond this PR. I'm basically poking around finding some easy way to point people in the right direction when they run into issues. Currently we have 300+ open issues but no process to even just triage them. Hence my suggestion for some sort of feature-specific pointer. Ideas welcome!

Reply:
`git blame` seems the way to go?  https://docs.github.com/en/github/creating-cloning-and-archiving-repositories/about-code-owners may be worth leveraging too.

Reply:
That's a significant barrier, X% of people will not jump it. Code tends to get moved around over time, then even of those X%, X+Y% will find the wrong person to "blame", or give up when they see the code was moved.
This PR is not the right place to solve the more general problem. I'm OK with this PR to go in without a pointer.